hass-addon-sunsynk-dev/run.py

```diff
@@ -49,18 +51,47 @@ async def publish_sensors(sensors: List[Filter], *, force: bool = False) -> None
         )
 
 
-async def hass_discover_sensors(serial: str) -> None:
+async def hass_discover_sensors(serial: str, rated_power: float) -> None:
     """Discover all sensors."""
     ents: List[Entity] = []
     dev = Device(
         identifiers=[OPT.sunsynk_id],
         name=f"Sunsynk Inverter {serial}",
-        model=f"Inverter {serial}",
+        model=f"{int(rated_power/1000)}kW Inverter {serial}",
         manufacturer="Sunsynk",
     )
 
+    def create_on_change_handler(filt: Filter, value_func: Callable):
+        async def _handler(value):
+            await write_sensor(filt, value_func(value))
```

Comment:
Maybe you can add some async debouncing here. 
With the profiles it only writes in the main loop and these are at least 1 second apart